File: sem2/OOP/PROJ 1/PO proj1/Guarana.cpp
#include "Guarana.h"

collision_result Guarana::collision(Organism* opponent) {
	opponent->change_strenght(GUARANA_EATING_BOOST); //guaran grants strenght boost to whoever ate it 
	world->log(opponent->get_species_name() + (string)" gains strenght ");
	return Plant::collision(opponent); // call base class collision, guarana gets eaten
}

File: sem2/OOP/PROJ 2/Kod/Guarana.py
from CollisionResult import CollisionResult
from Plant import Plant
import Constants


class Guarana(Plant):
    def __init__(self, position, world, creation_turn):
        super().__init__(Constants.GUARANA_STRENGTH, position, "guarana.png", world, creation_turn)

    def collision(self, opponent):
        result = super().collision(opponent)
        if result == CollisionResult.DIED:  # if guarana was eaten, boost strength of animal who ate it
            opponent.change_strength(Constants.GUARANA_STRENGTH_BOOST)
        return result

